parser/src/main/java/com/linkedin/avroutil1/writer/avsc/AvscSchemaWriter.java

```diff
@@ -399,6 +400,16 @@ protected JsonValue writeDefaultValue(AvroSchema fieldSchema, AvroLiteral litera
         AvroUnionSchema unionSchema = (AvroUnionSchema) fieldSchema;
         AvroSchema firstBranchSchema = unionSchema.getTypes().get(0).getSchema();
         return writeDefaultValue(firstBranchSchema, literal);
+      case RECORD:
+        AvroRecordSchema recordSchema = (AvroRecordSchema) fieldSchema;
+        JsonObjectBuilder recordObjectBuilder = Json.createObjectBuilder();
+        Map<String, AvroLiteral> recordLiteralMap = ((AvroRecordLiteral) literal).getValue();
+
+        for (AvroSchemaField field : recordSchema.getFields()) {
+          recordObjectBuilder.add(field.getName(),
```

Comment:
can you test this with a union field? i want to make sure it will correctly handle cases where the schema is union yet the value is a union branch